### PR TITLE
[latestVersionOfCommonTypesMustBeUsed] Fix null check

### LINKED ISSUE #619: [latestVersionOfCommonTypesMustBeUsed] Cannot read property 'split' of null
```
Error 1: $..['$ref'] threw: ErrorWithCause("Function \"latestVersionOfCommonTypesMustBeUsed\" threw an exception:
Cannot read property 'split' of null")
```

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3239434&view=logs&j=0574a2a6-2d0a-5ec6-40e4-4c6e2f70bea2&t=80c3e782-49f0-5d1c-70dd-cbee57bdd0c7&l=6404

Introduced in #492.

### PR DESCRIPTION
**Pull request by mikeharder**
- Fixes Azure/azure-sdk-tools#7428

### CODE REVIEW

**Review by bdefoy — APPROVED** [review]

**Comment by konrad-jamrozik** [comment]
@mikeharder just due diligence check: did you follow all the steps from here?
https://github.com/Azure/azure-openapi-validator/blob/main/CONTRIBUTING.md#how-to-prepare-for-a-pr-submission-after-you-made-changes-locally

File: packages/rulesets/src/spectral/functions/latest-version-of-common-types-must-be-used.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { LATEST_VERSION_BY_COMMON_TYPES_FILENAME, isLatestCommonTypesVersionForF
 export const latestVersionOfCommonTypesMustBeUsed = (ref: any, _opts: any, ctx: any) => {
   const REF_COMMON_TYPES_REGEX = new RegExp("/common-types/resource-management/v\\d+/\\w+.json#", "gi")
 
-  if (ref !== null && !ref.match(REF_COMMON_TYPES_REGEX)) {
+  if (ref === null || !ref.match(REF_COMMON_TYPES_REGEX)) {
```
Comment (konrad-jamrozik) [inline packages/rulesets/src/spectral/functions/latest-version-of-common-types-must-be-used.ts:6]: https://github.com/Azure/azure-openapi-validator/blob/main/CONTRIBUTING.md#create-a-release-pr

> Only update the version number for the relevant package(s) you want to deploy. For example, if your changes are made to the files under the [packages/rulesets](https://github.com/Azure/azure-openapi-validator/tree/main/packages/rulesets) directory, then update the version in [packages/rulesets/package.json](https://github.com/Azure/azure-openapi-validator/blob/main/packages/rulesets/package.json#L3). The same applies to all the other package directories in [packages directory](https://github.com/Azure/azure-openapi-validator/tree/main/packages).

**Comment by mikeharder** [comment]
> @mikeharder just due diligence check: did you follow all the steps from here? https://github.com/Azure/azure-openapi-validator/blob/main/CONTRIBUTING.md#how-to-prepare-for-a-pr-submission-after-you-made-changes-locally

No, just getting started with the PR.  Will convert to draft until I'm more ready.

**Review by timotheeguerin — APPROVED** [review]

**Review by rkmanda — APPROVED** [review]
:shipit:

**Comment by mikeharder** [comment]
I will get this PR ready to merge by EOD today.

**Pull request closed without merging**